source/common/router/router_ratelimit.cc

```diff
@@ -110,6 +110,29 @@ bool RemoteAddressAction::populateDescriptor(RateLimit::DescriptorEntry& descrip
   }
 
   descriptor_entry = {"remote_address", remote_address->ip()->addressAsString()};
+
+  return true;
+}
+
+bool MaskedRemoteAddressAction::populateDescriptor(RateLimit::DescriptorEntry& descriptor_entry,
+                                                   const std::string&,
+                                                   const Http::RequestHeaderMap&,
+                                                   const StreamInfo::StreamInfo& info) const {
+  const Network::Address::InstanceConstSharedPtr& remote_address =
+      info.downstreamAddressProvider().remoteAddress();
+  if (remote_address->type() != Network::Address::Type::Ip) {
+    return false;
+  }
+
+  uint32_t mask_len = v4_prefix_mask_len_;
+  if (remote_address->ip()->version() == Network::Address::IpVersion::v6) {
+    mask_len = v6_prefix_mask_len_;
+  }
+
+  Network::Address::CidrRange cidr_entry =
+      Network::Address::CidrRange::create(remote_address->ip()->addressAsString(), mask_len);
```

Comment:
This seems pretty inefficient. Can you at least add a TODO to increase the efficiency here? Seems like we can avoid a some string transform back and forth.